filter_plugins/redis.py

```diff
@@ -0,0 +1,68 @@
+        return {
+            'redis_master_address': self.redis_master_address,
+            'redis_master_port': self.redis_master_port,
+        }
+
+    def redis_master_address(self, fact, redis_type='redis', default_address=None, namespace='OPENIO', service=None, monitor_name='OPENIO-master-1'):
+        try:
+            if redis_type == 'redis':
+                return self._master_address_redis(fact, default_address=default_address, namespace=namespace, service=service)
+            elif redis_type == 'redissentinel':
+                return self._master_address_sentinel(fact, default_address=default_address, namespace=namespace, service=service, monitor_name=monitor_name)
+            else:
+                return default_address
+        except Exception as e:
+            return default_address
+
+    def _master_address_redis(self, fact, default_address=None, namespace='OPENIO', service=None):
+        if 'instances' in fact:
+            for instance in fact['instances']:
+                if instance['namespace'] == namespace and instance['service'] == service:
+                    if "master_host" in instance:
+                        return instance['master_host']
+
+        return default_address
+
+    def _master_address_sentinel(self, fact, default_address=None, namespace='OPENIO', service=None, monitor_name='OPENIO-master-1'):
+        if 'instances' in fact:
+            for instance in fact['instances']:
+                if instance['namespace'] == namespace and instance['service'] == service:
+                    for monitor in instance['monitors']:
+                        if "name" in monitor:
+                            if monitor["name"] == monitor_name:
+                                return monitor["master_host"]
+
+        return default_address
+
+    def redis_master_port(self, fact, redis_type='redis', default_port=6011, namespace='OPENIO', service=None, monitor_name='OPENIO-master-1'):
+        try:
+            if redis_type == 'redis':
+                return self._master_port_redis(fact, default_port=default_port, namespace=namespace, service=service)
+            elif redis_type == 'redissentinel':
+                return self._master_port_sentinel(fact, default_port=default_port, namespace=namespace, service=service, monitor_name=monitor_name)
+            else:
+                return default_port
+        except Exception as e:
+            return default_port
+
+    def _master_port_redis(self, fact, default_port=6011, namespace='OPENIO', service=None):
+        if 'instances' in fact:
+            for instance in fact['instances']:
+                if instance['namespace'] == namespace and instance['service'] == service:
+                    if "master_port" in instance:
+                        return instance['master_port']
+
+        return default_port
+
+    def _master_port_sentinel(self, fact, default_port=6011, namespace='OPENIO', service=None, monitor_name='OPENIO-master-1'):
```

Comment:
Eventually factor `_master_address_sentinel` and `_master_port_sentinel`? They're basically the same function. Find the master, return something, otherwise return the default.